gems/aws-sdk-core/lib/aws-sdk-core/plugins/jsonvalue_converter.rb

```diff
@@ -11,15 +11,31 @@ class Handler < Seahorse::Client::Handler
 
         def call(context)
           context.operation.input.shape.members.each do |m, ref|
-            if ref['jsonvalue']
-              param_value = context.params[m]
-              unless param_value.respond_to?(:to_json)
-                raise ArgumentError, "The value of params[#{m}] is not JSON serializable."
+            convert_jsonvalue(m, ref, context.params)
+          end
+          @handler.call(context)
+        end
+
+        def convert_jsonvalue(m, ref, params)
+          return if params.nil? || params[m].nil?
```

Comment:
We want to allow for serializing nil if it's provided explicitly. I think we will want to check params.key?(m) and if present, serialize it. Also can params ever be nil? It should always be at least a hash.

Reply:
Yes. And we need a test for not serializing "null" and doing so when deliberate.